assets/javascripts/modules/cookie-message.js

```diff
@@ -0,0 +1,44 @@
+/* global document */
+function setCookie(name, value, options = {}) {
+  let cookieString = `${name}=${value}; path=/`;
+
+  if (options.days) {
+    const date = new Date();
+    date.setTime(date.getTime() + (options.days * 24 * 60 * 60 * 1000));
+    cookieString = `${cookieString}; expires=${date.toGMTString()}`;
+  }
+
+  if (document.location.protocol === 'https:') {
```

Comment:
I share your concerns about cutting and pasting code. I can see what is going on but I'm not sure if this is good practice or not. I guess I'd opt for a minimal cookie package with testing and high usage but this comes with a cost of code bloat (e.g. https://github.com/js-cookie/js-cookie).